network/src/peer/mod.rs

```diff
@@ -64,27 +72,14 @@ pub enum DisconnectReason {
 
 impl fmt::Display for DisconnectReason {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(
-            f,
-            "{}",
-            match self {
-                DisconnectReason::Requested => "Requested",
-                DisconnectReason::ConnectionLost => "ConnectionLost",
-            }
-        )
+        let s = match self {
+            DisconnectReason::Requested => "Requested",
+            DisconnectReason::ConnectionLost => "ConnectionLost",
+        };
+        write!(f, "{}", s)
```

Comment:
I don't know if this was my fault earlier, but really this should probably be the `as_str` then `display` functionality.  But, let's not bother changing that.